packages/tools/src/lib/build-cache/common.ts

```diff
@@ -10,28 +10,79 @@ export type SupportedRemoteCacheProviders = 'github-actions';
 
 export type RemoteArtifact = {
   name: string;
-  downloadUrl: string;
+  url: string;
+  id?: string;
 };
 
 export type LocalArtifact = {
   name: string;
   path: string;
 };
 
+/**
+ * Interface for implementing remote build cache providers.
+ * Remote cache providers allow storing and retrieving native build artifacts (e.g. APK, IPA)
+ * from remote storage like S3, GitHub Artifacts etc.
+ */
 export interface RemoteBuildCache {
+  /** Unique identifier for this cache provider, will be displayed in logs */
   name: string;
-  query({
+
+  /**
+   * List available artifacts matching the given name pattern
+   * @param artifactName - Passed after fingerprinting the build, e.g. `rnef-android-debug-1234567890` for android in debug variant
+   * @param limit - Optional maximum number of artifacts to return
+   * @returns Array of matching remote artifacts, or null if none found
+   */
+  list({
     artifactName,
+    limit,
   }: {
-    artifactName: string;
-  }): Promise<RemoteArtifact | null>;
+    artifactName: string | undefined;
+    limit?: number;
+  }): Promise<RemoteArtifact[] | null>;
+
+  /**
+   * Download a remote artifact to local storage
+   * @param artifact - Remote artifact to download, as returned by `list` method
+   * @param loader - Optional progress indicator
+   * @returns Local artifact info after download
+   */
   download({
     artifact,
     loader,
   }: {
     artifact: RemoteArtifact;
-    loader: ReturnType<typeof spinner>;
+    loader?: ReturnType<typeof spinner>;
```

Comment:
since `loader` is passed by the cli itself, when would it be optional? is it possible to pass a mocked loader instead? it'll be nice if the implementations didn't need to worry about conditionally handling loader.

Reply:
E.g. in https://github.com/callstack/rnef/pull/263 I'm prototyping a `remote-cache` command with raw output (so I can run `ARTIFACT=$(npx rnef remote-cache download {artifactName} --json)` on any CI to simplify integrating RNEF outside of GHActions. When passing the `--json` flag (not implemented yet) I would omit passing the loader so that the output is not obscured.

Reply:
@thymikee how about

> is it possible to pass a mocked loader instead?

in the end the cli owns the loader object, so whether it prints anything is up to the cli. so it'd be better if cli handled it automatically instead of requiring cache providers to handle it.